From: Gregory Price <gregory.price@memverge.com>
Subject: mm-mempolicy-introduce-mpol_weighted_interleave-for-weighted-interleaving-fix.
Date: Wed, 31 Jan 2024 00:12:24 -0500

kill next_node in favor of operating directly on il_prev

Link: https://lkml.kernel.org/r/ZbnWuB4dRCEFRz2m@memverge.com
Signed-off-by: Gregory Price <gregory.price@memverge.com>
Cc: Andi Kleen <ak@linux.intel.com>
Cc: Dan Williams <dan.j.williams@intel.com>
Cc: Frank van der Linden <fvdl@google.com>
Cc: Hasan Al Maruf <Hasan.Maruf@amd.com>
Cc: Honggyu Kim <honggyu.kim@sk.com>
Cc: Huang Ying <ying.huang@intel.com>
Cc: Hyeongtak Ji <hyeongtak.ji@sk.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Jonathan Cameron <Jonathan.Cameron@huawei.com>
Cc: Jonathan Corbet <corbet@lwn.net>
Cc: Michal Hocko <mhocko@kernel.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Rakie Kim <rakie.kim@sk.com>
Cc: Ravi Jonnalagadda <ravis.opensrc@micron.com>
Cc: Srinivasulu Thanneeru <sthanneeru.opensrc@micron.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mempolicy.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/mempolicy.c~mm-mempolicy-introduce-mpol_weighted_interleave-for-weighted-interleaving-fix
+++ a/mm/mempolicy.c
@@ -2378,7 +2378,7 @@ static unsigned long alloc_pages_bulk_ar
 	unsigned int weight_total = 0;
 	unsigned long rem_pages = nr_pages;
 	nodemask_t nodes;
-	int nnodes, node, next_node;
+	int nnodes, node;
 	int resume_node = MAX_NUMNODES - 1;
 	u8 resume_weight = 0;
 	int prev_node;
@@ -2408,7 +2408,7 @@ static unsigned long alloc_pages_bulk_ar
 		} else if (rem_pages == weight) {
 			/* move to next node / weight */
 			me->il_prev = next_node_in(node, nodes);
-			me->il_weight = get_il_weight(next_node);
+			me->il_weight = get_il_weight(me->il_prev);
 			return total_allocated;
 		}
 		/* Otherwise we adjust remaining pages, continue from there */
_
